### DISCUSSION TITLE
Improvement of modules

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1278] [General]
Today I am talking about, mutually recursive structures using this 'and' keyword is possible within the same module, but not if they are in different files + modules
A lot people create a file for a type so would be nice if we can automatically (like C#) or manually (via 'and) get to use the type we created in another file

**Comment by Happypig375** [comment]
One way to implement this is to specify mutual recursion in the `fsproj` and have visual studio show the mutual recursion. However this brings too much complexity for little value. Just keep your mutual recursions in the same file.

**Reply by Xyncgas** [reply]
Actually, you are suggesting something useful. I don't change files often after I defined my types. Yet I know not methods to achieve this, perhaps if you may elaborate further on the method you hath proposed, it would be greatly appreciated

**Reply by Happypig375** [reply]
I meant something like this in the fsproj file
```xml
<Compile Include="a.fs" />
<Compile Include="b.fs" RecurseWith="a.fs" />
<Compile Include="c.fs" RecurseWith="a.fs" />
```
making `a`, `b`, `c` all mutually recursive.

But this brings a lot of implementation complexity to the compiler while encouraging poor code structure with excessive mutual recursion. It wouldn't be implemented.

**Reply by Xyncgas** [reply]
Thanks.
